agent/05_agent_create_cluster.sh

```diff
@@ -96,6 +97,51 @@ function wait_for_cluster_ready() {
   echo "Cluster is ready!"
 }
 
+function mce_prepare_postinstallation_manifests() {
+  local mceManifests=$1
+
+  # Copy all the manifests required after the installation completed
+  cp ${SCRIPTDIR}/agent/assets/mce/agent_mce_1*.yaml ${mceManifests}
+
+  # Render the cluster image set template
+  local clusterImageSetTemplate=${mceManifests}/agent_mce_1_04_clusterimageset.yaml
+  local version="$(openshift_version ${OCP_DIR})"
+  local releaseImage=$(getReleaseImage)
+
+  sed -i "s/<version>/${version}/g" ${clusterImageSetTemplate}
+  sed -i "s/<releaseImage>/${releaseImage//\//\\/}/g" ${clusterImageSetTemplate}
+}
+
+function mce_apply_postinstallation_manifests() {
+  local mceManifests=$1
+
+  wait_for_crd "localvolumes.local.storage.openshift.io"
+  apply_manifest "$mceManifests/agent_mce_1_01_localvolumes.yaml"
+  oc wait localvolume -n openshift-local-storage assisted-service --for condition=Available --timeout 10m || exit 1
+
+  wait_for_crd "multiclusterengines.multicluster.openshift.io"
+  apply_manifest "$mceManifests/agent_mce_1_02_mce.yaml"
+
+  wait_for_crd "agentserviceconfigs.agent-install.openshift.io"
+  apply_manifest "$mceManifests/agent_mce_1_03_agentserviceconfig.yaml"
+
+  wait_for_crd "clusterimagesets.hive.openshift.io"
+  apply_manifest "$mceManifests/agent_mce_1_04_clusterimageset.yaml"
+
+  apply_manifest "$mceManifests/agent_mce_1_05_autoimport.yaml"
+  oc wait -n multicluster-engine managedclusters local-cluster --for condition=ManagedClusterJoined=True --timeout 10m || exit 1
```

Comment:
In my experience the `wait_for_crd` just slows down the job. We can keep it to see the impact. Since Openshift is fully running by this point you should be able to batch create all of these and then check for the volume and cluster. It will take it some time to reconcile but it should reconcile.

Reply:
Right, I left them (not batched) mostly for the sake of clarity and just to be sure that the crds where available - they should be already present as you pointed out, so the overall crds checks should be pretty quick